### PR TITLE
Add exercise: phone-number

### PR DESCRIPTION
**Pull request by Teo-ShaoWei**
This is a call-to-arm to issue #87. First time here, pardon me if anything is out of spec. Feel free to inform me on how I can polish it and I will get it fixed.

Some design decisions...
- The main machinery of this question is regex. Instead of mixing up regex with language specifics like some other language tracks, I decided to put the full power on regex. Instead the example demonstrates how Julia can call regex, and create regex using interpolation to improve code readability.
- I did not put in type declaration for the input variable, pending discussion on issue #2. I will adhere to the convention of this repo after we concluded the discussion.

### CODE REVIEW

**Comment by Teo-ShaoWei** [comment]
Thanks for the review @SaschaMann =)

Yupz I agree with you that this would be a floating non-core exercise. I think I left the `config.json` entry of this problem as `"core": false,`. Is that the correct setting?

Exercism v2? That's new to me. Where can I find out more about it?

File: exercises/phone-number/example.jl [hunk 1]
```diff
@@ -0,0 +1,18 @@
+function clean(phone_number)
+    # Main machinery is in the regex expression.
+    NXX = "[2-9][\\d]{2}"
+    country = "(?:\\+?1)?"
+    area = "\\(?\\s*($NXX)\\s*\\)?"
+    exchange = "($NXX)"
+    subscriber = "([\\d]{4})"
+    fillers = "\\s*(?:\\.|-)?\\s*"
+    r = Regex("^\\s*$country$fillers$area$fillers$exchange$fillers$subscriber\\s*\$")
```
[the remark this answers is not in the document]
Comment (Teo-ShaoWei) [reply exercises/phone-number/example.jl]: Yes I would have preferred to demonstrate regex string literals but I think it is not possible to do regex interpolation (refer to this [issue](https://github.com/JuliaLang/julia/issues/1797)). At the end of it the suggested solution is to use the `Regex` constructor (as per this [discourse](https://discourse.julialang.org/t/regex-w-string-interpolation/3545/2)). The string looks more unwieldy without interpolation, so I took the lesser of the two evils.

Another way is to extract all the digits first and then check only the digits, but there will be a lot of false positives.

I am not good with regex myself too actually haha... I think the test cases are ok, so we can consider publish it first, and then we might have a submission that solve this nicely =)

**Comment by SaschaMann** [comment]
Yep, that's the correct setting.

Info about exercism v2 is spread out over the discussion and docs repos. One of the major changes is the progression tree as opposed to the current, linear progression.

Some info: 
https://github.com/exercism/discussions/issues/175 
https://github.com/exercism/discussions/issues/159
https://github.com/exercism/discussions/issues/154

The prototype/beta is available here: v2.exercism.io (Julia isn't deployed on that yet)

File: exercises/phone-number/example.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (SaschaMann) [reply exercises/phone-number/example.jl]: Ah okay, I didn't know of that.

**Comment by SaschaMann** [comment]
Thanks for adding this! :)

**Comment by Teo-ShaoWei** [comment]
Ah I will give Exercism v2 info a good read. It will be good if we can prepare for it while we import questions.

Thanks for reviewing too. I look forward to working with you and the rest of the team on other imports =)

**Pull request merged**